transaction.go

```diff
@@ -4877,6 +4877,27 @@ func (tx *Transaction) execute(client *Client, e TransactionInterface) (Transact
 	}, nil
 }
 
+func (tx *Transaction) FreezeWith(client *Client, e TransactionInterface) (TransactionInterface, error) { //nolint
```

Comment:
TODO: we need a way of calling freeze on a transaction, without knowing the concrete type of the transaction. It's used in `FillOutTransaction` method

Reply:
Could it be achievable if `FillOutTransaction` is a method of the Transactions themselves?

Reply:
I don't see the point, as `FillOutTransaction` would not fit in any way in the existing flows of the sdk. 
I just need a way to expose the freezeWith function, as it's exposed in Java.

Reply:
We can take a look. Maybe I don't have enough context.